docs: various (minor) changes and rewording (#3661)
docker/cli: docs/reference/commandline/attach.md

> so.

It is forbidden to redirect the standard input of a `docker attach` command

> while attaching to a TTY-enabled container (i.e., launched with `-i` and `-t`).

Reviewer: WDYT about losing the i.e. for readability, I think it is implied in a aside like: "(using the `-i` or `-t` options)" or even "(`-i`, `-t` options).

> While a client is connected to container's stdio using `docker attach`, Docker

Reviewer: ```suggestion
While a client is connected to container's `stdio` using `docker attach`, Docker
```
Considering it's a file.

uses a ~1MB memory buffer to maximize the throughput of the application. If

> this buffer is filled, the speed of the API connection will have an
> effect on the process output writing speed. This is similar to other
> applications like SSH. Because of this, it is not recommended to run
> performance critical applications that generate a lot of output in the

Reviewer: (After removing the "If" in line 54,)
```suggestion
Once this buffer is full, the speed of the API connection is affected, and so this impacts the output process' writing speed. 
```

### Attach to and detach from a running container

The following example starts an ubuntu container running `top` in detached mode,
then attaches to the container;

```console
$ docker run -d --name topdemo ubuntu:22.04 /usr/bin/top -b

$ docker attach topdemo

top - 12:27:44 up 3 days, 21:54,  0 users,  load average: 0.00, 0.00, 0.00
Tasks:   1 total,   1 running,   0 sleeping,   0 stopped,   0 zombie
%Cpu(s):  0.1 us,  0.1 sy,  0.0 ni, 99.8 id,  0.0 wa,  0.0 hi,  0.0 si,  0.0 st
MiB Mem :   3934.3 total,    770.1 free,    674.2 used,   2490.1 buff/cache
MiB Swap:   1024.0 total,    839.3 free,    184.7 used.   2814.0 avail Mem

  PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND
    1 root      20   0    7180   2896   2568 R   0.0   0.1   0:00.02 top
```

> As the container was started without the `-i`, and `-t` options, signals are

Reviewer: Do we need to use -i and -t in conjunction? Or is the case of using one or the other?

> forwarded to the attached process, which means that the default `CTRL-p CTRL-q`
> detach key sequence takes no effect, and pressing `CTRL-c` will terminate the
> container:

Reviewer: ```suggestion
forwarded to the attached process, which means that the default `CTRL+P`,`CTRL+Q`
detach key sequence produces no effect, but pressing `CTRL+C` terminates the
container:
```

```console
<...>
  PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND
    1 root      20   0    7180   2896   2568 R   0.0   0.1   0:00.02 top^P^Q
^C

$ docker ps -a --filter name=topdemo

CONTAINER ID   IMAGE          COMMAND             CREATED              STATUS                          PORTS     NAMES
4cf0d0ebb079   ubuntu:22.04   "/usr/bin/top -b"   About a minute ago   Exited (0) About a minute ago             topdemo
```

Repeating the example above, but this time with the `-i` and `-t` options set;

```console
$ docker run -dit --name topdemo2 ubuntu:22.04 /usr/bin/top -b
```

> Now, when attaching to the container, and pressing the `CTRL-p CTRL-q` sequence,
> the Docker CLI is handling the detach sequence ("read escape sequence"), and the
> `attach` command is detached from the container. Checking the container's status
> with `docker ps` shows that the container is still running in the background:

Reviewer: ```suggestion
Now, when attaching to the container, and pressing the `CTRL+P CTRL+Q` ("read escape sequence"),
the Docker CLI is handling the detach sequence, and the
`attach` command is detached from the container. Checking the container's status
with `docker ps` shows that the container is still running in the background:
```
